cmd/serve/main.go

```diff
@@ -1,22 +1,53 @@
 package main
 
 import (
+	"context"
 	"embed"
 	"fmt"
 	"io/fs"
 	"log"
 	"net/http"
+	"os"
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq"
+
+	create_user_v1 "github.com/sunb26/xat/handler/create_user"
 )
 
 //go:embed all:web all:web/_next
 var content embed.FS
 
+type middleware struct {
+	handler http.Handler
+	db      *sqlx.DB
+}
+
+func (i *middleware) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx := context.WithValue(r.Context(), "db", i.db)
+	i.handler.ServeHTTP(w, r.WithContext(ctx))
+}
+
+func newMiddleware(handler http.Handler, db *sqlx.DB) *middleware {
+	return &middleware{handler: handler, db: db}
+}
+
 func main() {
 	content, err := fs.Sub(content, "web")
 	if err != nil {
 		log.Fatal(err)
 	}
-	http.Handle("/", http.FileServerFS(content))
+
+	db, err := sqlx.Open("postgres", os.Getenv("DSN"))
```

Comment:
replace with connect? https://stackoverflow.com/questions/52318684/what-is-the-difference-between-sqlx-connect-and-sqlx-open